src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -7565,21 +7509,11 @@ namespace WaterThermalTanks {
 				if (MaxSpeedNum > 0) {
 					//square the solving, and avoid warning
 					//due to very small capacity at lowest speed of VSHPWH coil
-					{ auto const SELECT_CASE_var1(HeatPump.TankTypeNum);
-					if (SELECT_CASE_var1 == MixedWaterHeater) {
-						zeroResidual = PLRResidualMixedTank(0.0, Par);
-					} else if (SELECT_CASE_var1 == StratifiedWaterHeater) {
-						zeroResidual = PLRResidualStratifiedTank(0.0, Par);
-					}}
+					zeroResidual = PLRResidualHPWH( 0.0, Par );
 				}
 
 				if (zeroResidual > 0.0) { // then iteration
-					{ auto const SELECT_CASE_var1(HeatPump.TankTypeNum);
-					if (SELECT_CASE_var1 == MixedWaterHeater) {
-						SolveRegulaFalsi(Acc, MaxIte, SolFla, HPPartLoadRatio, PLRResidualMixedTank, 0.0, 1.0, Par);
-					} else if (SELECT_CASE_var1 == StratifiedWaterHeater) {
-						SolveRegulaFalsi(Acc, MaxIte, SolFla, HPPartLoadRatio, PLRResidualStratifiedTank, 0.0, 1.0, Par);
-					}}
+					SolveRegulaFalsi( Acc, MaxIte, SolFla, HPPartLoadRatio, PLRResidualHPWH, 0.0, 1.0, Par );
```

Comment:
I combined the heat pump water heater PLR residual functions into one for simplicity in managment.